focus-android/app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt

```diff
@@ -50,7 +50,24 @@ class BrowserRobot {
         )
 
     fun verifyPageContent(expectedText: String) {
-        mDevice.wait(Until.findObject(By.textContains(expectedText)), waitingTime)
+        sessionLoadedIdlingResource = SessionLoadedIdlingResource()
+
+        runWithIdleRes(sessionLoadedIdlingResource) {
+            for (i in 1..RETRY_COUNT) {
+                try {
+                    assertTrue(
+                        webPageItemContainingText(expectedText).waitForExists(pageLoadingTime),
+                    )
+                    break
+                } catch (e: AssertionError) {
+                    if (i == RETRY_COUNT) {
+                        throw e
+                    } else {
+                        refreshPageIfStillLoading(expectedText)
+                    }
+                }
+            }
+        }
```

Comment:
Truthfully I'm confused between the difference between  `runWithIdleRes` and how we use other `IdlingResources` still.  ` runWithIdleRes` is doing it's own check above and `sessionLoadedIdlingResource` is doing a check [here](https://github.com/mozilla-mobile/fenix/blob/6a3adb5aeba976a7146eb96a2ef3794fae909304/app/src/androidTest/java/org/mozilla/fenix/helpers/SessionLoadedIdlingResource.kt#L30). What is the difference?

Reply:
runWithIdleRes is just registering and unregistering the resources given. In this case, the sessionIdlingRes.